packages/@aws-cdk/aws-ec2/lib/private/ebs-util.ts

```diff
@@ -24,19 +24,39 @@ function synthesizeBlockDeviceMappings<RT, NDT>(construct: Construct, blockDevic
   return blockDevices.map<RT>(({ deviceName, volume, mappingEnabled }): RT => {
     const { virtualName, ebsDevice: ebs } = volume;
 
+    let finalEbs: any;
+
     if (ebs) {
-      const { iops, volumeType } = ebs;
 
+      const { iops, volumeType } = ebs;
       if (!iops) {
         if (volumeType === EbsDeviceVolumeType.IO1) {
           throw new Error('iops property is required with volumeType: EbsDeviceVolumeType.IO1');
         }
       } else if (volumeType !== EbsDeviceVolumeType.IO1) {
         Annotations.of(construct).addWarning('iops will be ignored without volumeType: EbsDeviceVolumeType.IO1');
       }
+
+      /**
+       * Because the Ebs properties of the L2 Constructs do not match the Ebs properties of the Cfn Constructs,
+       * we have to do some transformation.
+       */
+
+      finalEbs = {};
+
+      for ( var propertie in ebs ) {
+        if ( propertie === 'kmsKey' ) {
+          finalEbs.kmsKeyId = ebs[propertie]?.keyArn;
+        } else {
+          finalEbs[propertie] = (ebs as any)[propertie];
+        }
+      }
```

Comment:
This reads much simpler to me.

```suggestion
      ebsProperty = {
        ...ebs,
        kmsKeyId: ebs.kmsKey?.keyArn,
      };
```

Reply:
yes, but this way won't work. The field kmsKey won't be removed, but there is the ...rest destructing operator, ... I have an Idea

Reply:
```
let finalEbs:  CfnLaunchTemplate.EbsProperty | CfnInstance.EbsProperty | undefined;

    if (ebs) {

      const { iops, volumeType, kmsKey, ...unchanged } = ebs;
      if (!iops) {
        if (volumeType === EbsDeviceVolumeType.IO1) {
          throw new Error('iops property is required with volumeType: EbsDeviceVolumeType.IO1');
        }
      } else if (volumeType !== EbsDeviceVolumeType.IO1) {
        Annotations.of(construct).addWarning('iops will be ignored without volumeType: EbsDeviceVolumeType.IO1');
      }

      /**
       * Because the Ebs properties of the L2 Constructs do not match the Ebs properties of the Cfn Constructs,
       * we have to do some transformation.
       */


      finalEbs = {
        ...unchanged,
        kmsKeyId: kmsKey?.keyArn,
      };

    } else {
      finalEbs = undefined;
    }
```
Think this is the best solution?